build/copyright-check.sh

```diff
@@ -0,0 +1,183 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Copyright Contributors to the Open Cluster Management project
+
+
+#Project start year
+origin_year=2016
+#Back up year if system time is null or incorrect
+back_up_year=2019
+#Currrent year
+current_year=$(date +"%Y")
+
+TRAVIS_BRANCH=$1
+
+ADDED_SINCE_1_MAR_2020=$(git log --name-status --pretty=oneline --since "1 Mar 2020" | egrep "^A\t" | awk '{print $2}' | sort | uniq |  grep -v -f <(sed 's/\([.|]\)/\\\1/g; s/\?/./g ; s/\*/.*/g' .copyrightignore))
+MODIFIED_SINCE_1_MAR_2020=$(diff --new-line-format="" --unchanged-line-format="" <(git log --name-status --pretty=oneline --since "1 Mar 2020" | egrep "^A\t|^M\t" | awk '{print $2}' | sort | uniq | grep -v -f <(sed 's/\([.|]\)/\\\1/g; s/\?/./g ; s/\*/.*/g' .copyrightignore)) <(git log --name-status --pretty=oneline --since "1 Mar 2020" | egrep "^A\t" | awk '{print $2}' | sort | uniq | grep -v -f <(sed 's/\([.|]\)/\\\1/g; s/\?/./g ; s/\*/.*/g' .copyrightignore)))
+OLDER_GIT_FILES=$(git log --name-status --pretty=oneline | egrep "^A\t|^M\t" | awk '{print $2}' | sort | uniq |  grep -v -f <(sed 's/\([.|]\)/\\\1/g; s/\?/./g ; s/\*/.*/g' .copyrightignore))
+
+if [[ "x${TRAVIS_BRANCH}" != "x" ]]; then
+  FILES_TO_SCAN=$(git diff --name-only --diff-filter=AM ${TRAVIS_BRANCH}...HEAD | grep -v -f <(sed 's/\([.|]\)/\\\1/g; s/\?/./g ; s/\*/.*/g' .copyrightignore))
+else
+  FILES_TO_SCAN=$(find . -type f | grep -Ev '(\.git)' | grep -v -f <(sed 's/\([.|]\)/\\\1/g; s/\?/./g ; s/\*/.*/g' .copyrightignore))
+fi
+
+if [ -z "$current_year" ] || [ $current_year -lt $origin_year ]; then
+  echo "Can't get correct system time\n  >>Use back_up_year=$back_up_year as current_year to check copyright in the file $f\n"
+  current_year=$back_up_year
+fi
+
+lic_ibm_identifier=" (c) Copyright IBM Corporation"
+lic_redhat_identifier_2020=" Copyright (c) 2020 Red Hat, Inc."
+lic_redhat_identifier_2021=" Copyright (c) ${current_year} Red Hat, Inc."
+
+lic_year=()
+#All possible combination within [origin_year, current_year] range is valid format
+#seq isn't recommanded after bash version 3.0
+for ((start_year=origin_year;start_year<=current_year;start_year++)); 
+do
+  lic_year+=(" (c) Copyright IBM Corporation ${start_year}. All Rights Reserved.")
+  for ((end_year=start_year+1;end_year<=current_year;end_year++)); 
+  do
+    lic_year+=(" (c) Copyright IBM Corporation ${start_year}, ${end_year}. All Rights Reserved.")
+  done
+done
+lic_year_size=${#lic_year[@]}
+
+#lic_rest to scan for rest copyright format's correctness
+lic_rest=()
+lic_rest+=(" Licensed Materials - Property of IBM")
+lic_rest+=(" Note to U.S. Government Users Restricted Rights:")
+lic_rest+=(" Use, duplication or disclosure restricted by GSA ADP Schedule")
+lic_rest+=(" Contract with IBM Corp.")
+lic_rest_size=${#lic_rest[@]}
```

Comment:
We'll need to clean up this copyright check eventually (and we may not even need it?), but that's an effort for another day, I think